action.yaml

```diff
@@ -79,14 +79,37 @@ inputs:
     description: Whether to register the output paths of each flake output with FlakeHub.
     required: false
 
+  # Used to construct the binary download URL
+  source-binary:
+    description: Run a version of the `flakehub-push` binary from somewhere already on disk. Conflicts with all `source-*` options.
+    required: false
+  source-branch:
+    description: The branch of `flakehub-push` to use. Conflicts with all other `source-*` options.
+    required: false
+    default: main
+  source-pr:
+    description: The pull request (PR) of `flakehub-push` to use. Conflicts with all other `source-*` options.
+    required: false
+  source-revision:
+    description: The revision of `flakehub-push` to use. Conflicts with all other `source-*` options.
+    required: false
+  source-tag:
+    description: The tag of `flakehub-push` to use. Conflicts with all other `source-*` options.
+    required: false
+  source-url:
+    description: A URL pointing to a `flakehub-push` binary. Overrides all other `source-*` options.
+    required: false
+
+  # DEPRECATED options (use the `source-*` equivalents above)
   flakehub-push-binary:
-    description: Run a version of the `flakehub-push` binary from somewhere already on disk. Conflicts with all other `flakehub-push-*` options.
```

Comment:
Looks like we'll need to drop the default values here, as they're being detected as using deprecated options.